doc/pages/DEV_DOC.md

```diff
@@ -0,0 +1,331 @@
+run).
+
+Supported environment variables are described in the following list:
+
+*   `BENCHMARK={spmv, solver, preconditioner}` - The benchmark set to run.
+    Default is `spmv`.
+    *   `spmv` - Runs the sparse matrix-vector product benchmarks on the
+                 SuiteSparse collection.
+    *   `solver` - Runs the solver benchmarks on the SuiteSparse collection.
+                The matrix format is determined by running the `spmv` benchmarks
+                first, and using the fastest format determined by that
+                benchmark. The maximum number of iterations for the iterative
+                solvers is set to 10,000 and the requested residual reduction
+                factor to 1e-6.
+    *   `preconditioner` - Runs the preconditioner benchmarks on artificially
+                generated block-diagonal matrices.
+*   `DRY_RUN={true, false}` - If set to `true`, prepares the system for the
+    benchmark runs (downloads the collections, creates the result structure,
+    etc.) and outputs the list of commands that would normally be run, but does
+    not run the benchmarks themselves. Default is `false`.
+*   `EXECUTOR={reference,cuda,omp}` - The executor used for running the
+    benchmarks. Default is `cuda`.
+*   `SEGMENTS=<N>` - Splits the benchmark suite into `<N>` segments. This option
+    is useful for running the benchmarks on an HPC system with a batch
+    scheduler, as it enables partitioning of the benchmark suite and running it
+    concurrently on multiple nodes of the system. If specified, `SEGMENT_ID`
+    also has to be set. Default is `1`.
+*   `SEGMENT_ID=<I>` - used in combination with the `SEGMENTS` variable. `<I>`
+    should be an integer between 1 and `<N>`. If specified, only the `<I>`-th
+    segment of the benchmark suite will be run. Default is `1`.
+*   `SYSTEM_NAME=<name>` - the name of the system where the benchmarks are being
+    run. This option only changes the directory where the benchmark results are
+    stored. It can be used to avoid overwriting the benchmarks if multiple
+    systems share the same filesystem, or when copying the results between
+    systems. Default is `unknown`.
+
+Once `make benchmark` completes, the results can be found in
+`<Ginkgo build directory>/benchmark/results/${SYSTEM_NAME}/`. The files are
+written in the JSON format, and can be analyzed using any of the data
+analysis tools that support JSON. Alternatively, they can be uploaded to an
+online repository, and analyzed using Ginkgo's free web tool
+[Ginkgo Performance Explorer (GPE)](https://ginkgo-project.github.io/gpe/).
+(Make sure to change the "Performance data URL" to your repository if using
+GPE.)
+
+@page Contribution guidelines
+
+Do you want to contribute to Ginkgo ? We would be happy to help you get started. Please have a look at
+the [contribution guidelines](https://github.com/ginkgo-project/ginkgo/wiki/Contributing-guidelines).
+
```

Comment:
Sorry, that was my fault. I did not remove the contents of this and WHITEPAPER.md files. I actually added the automatic creation from the BASE_DOC.md file just a concatenation using CMake. But forgot to remove the contents of this file.

The reason for 3 separate files is so that is there is anything that would be separate in the three different things they could go into their specific pages. Currently only the first line that describes what kind of documentation this is varies, but more stuff/explanations can be added.

Reply:
OK, I just saw for the `concat`. That is better as it gets down the duplication to just one, but again I would rather go with a solution where `BASE_DOC` itself is made from directly using the main markdown files from our repository. Because that is documentation duplication and a very easy point of failure.

Reply:
Yes, I am doing that now.